README.md

```diff
@@ -3,8 +3,99 @@
 
 ## Goal
 
+> `v1` documentation below
+
 Validation of kubernetes templates and also a rich diff to state in cluster.
 
+## Usage
+
+The workflow uses the kubernetes engineering image to run the ci steps. The
+image auto detects kubernetes objects required validating and can also produce
+a diff in case of changes. To use the this in CI, setup the workflow like this:
+
+```yaml
+  kubernetes-ci:
+    name: "Kubernetes CI "
+    concurrency:
+      group: ${{ github.repository }}-${{ github.workflow }}-kubernetes-ci-${{ github.ref }}
+      cancel-in-progress: true
+    needs: ["setup"]
+    if: ${{ needs.setup.outputs.run-kubernetes-ci == 'true'}}
+    uses: coopnorge/github-workflow-kubernetes-validation/.github/workflows/kubernetes-validation.yaml@v2.0.0
+    secrets:
+      argocd-api-token: ${{ secrets.ARGOCD_API_TOKEN }}
+```
+
+And have in your `docker-compose.yaml`
+
+```yaml
+  kubernetes-devtools:
+    build:
+      context: docker-compose
+      target: kubernetes-devtools
+      dockerfile: Dockerfile
+    privileged: false
+    command: validate
+    security_opt:
+      - seccomp:unconfined
+      - apparmor:unconfined
+    volumes:
+      - .:/srv/workspace:z
+      - $HOME/.argocd:/root/.config/argocd
+```
+
+and in `docker-compose/Dockerfile` have atleast this image
+
+```dockerfile
+FROM ghcr.io/coopnorge/engineering-docker-images/e0/devtools-kubernetes-v1beta1:latest@sha256:db0e83c6ae634f27c14c1b3ff55e1b12d94066222380b1cada65ae72023b5fc6 AS kubernetes-devtools
+```
+
+Make sure you update your dependabot to update docker images in `docker-compose/Dockerfile`
+
+## Inputs
+
+```yaml
+    inputs:
+      do-argocd-diff:
+        required: false
+        default: true
+        type: boolean
+        description: |
+          Boolean to enable argocd diff.
+      docker-compose-file:
+        type: string
+        default: ./docker-compose.yaml
+        required: false
+        description: |
+          Location of the docker-compose file relative to the root of the repository
+      docker-compose-service:
+        type: string
+        default: kubernetes-devtools
+        required: false
+        description: |
+          The name of the service in the docker-compose file to use for validating terraform
+      argocd-server:
+        type: string
+        default: argocd.internal.coop
+        required: false
+        description: |
+          The dns server of ArgoCD. Defaults to 'argocd.internal.coop'
+      validate-target:
+        type: string
+        default: validate
+        required: false
+        description: |
+          Target to run when running the validate step, defaults to 'validate'
+
+    secrets:
+      argocd-api-token:
+        required: false
+        description: |
+          Token to get read only access to argocd. This is a global secret secrets.ARGOCD_API_TOKEN
+```
```

Comment:
I think we should have end user docs in `docs/index.md` so that it is published to Inventory.

Reply:
https://github.com/coopnorge/github-workflow-kubernetes-validation/issues/36

Reply:
@nhhagen can you resolve?